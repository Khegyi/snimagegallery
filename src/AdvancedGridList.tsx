/* eslint-disable require-jsdoc */
import React from 'react'
import { User } from '@sensenet/default-content-types'
import { IconButton } from '@material-ui/core'
import GridList from '@material-ui/core/GridList'
import GridListTile from '@material-ui/core/GridListTile'
import GridListTileBar from '@material-ui/core/GridListTileBar'
import { useRepository } from './hooks/use-repository'
import { useStyles } from './app'
import { AdvancedGridprops } from './Interface'

export function AdvancedGridList(props: AdvancedGridprops) {
  const classes = useStyles()
  const repo = useRepository()

  function pickTile(anumber: any) {
    let tilenumber = anumber % 3
    tilenumber = tilenumber === 0 ? 2 : 1
    return tilenumber
  }
  return (
    <div className={classes.root}>
      <GridList cellHeight={200} spacing={1} className={classes.gridList}>
        {props.imgList.map((tile, index) => (
          <GridListTile key={tile.Id} cols={pickTile(index)} rows={pickTile(index)}>
            <img
              className={classes.imgTile}
              src={repo.configuration.repositoryUrl + tile.Path}
              onClick={() => props.openFunction(index, true)}
              alt={tile.Description}
            />
            <GridListTileBar
              title={tile.DisplayName}
              titlePosition="bottom"
              subtitle={<span>by: {(tile.CreatedBy as User).FullName}</span>}
              actionIcon={<IconButton aria-label={`star ${tile.DisplayName}`} className={classes.icon}></IconButton>}
              actionPosition="left"
              className={classes.titleBar}
            />
          </GridListTile>
        ))}
      </GridList>
    </div>
  )
}
