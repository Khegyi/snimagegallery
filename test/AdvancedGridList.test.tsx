import { shallow } from 'enzyme'
import React from 'react'
import { AdvancedGridList, pickTile } from '../src/components/AdvancedGridList'

describe('AdvancedGridList', () => {
  const testprops = {
    openFunction: jest.fn(),
    imgList: [],
  }
  it('Matches snapshot', () => {
    const wrapper = shallow(<AdvancedGridList {...testprops} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('pickTile', () => {
    it('should return 1 or 2', () => {
      const value = pickTile(456)
      expect(value).toBe(2)
    })
  })
})
