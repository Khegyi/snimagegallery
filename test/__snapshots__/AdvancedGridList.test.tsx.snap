// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedGridList Matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(GridList))
    cellHeight={200}
    className="makeStyles-gridList-2"
    spacing={1}
  />
</div>
`;
